### ISSUE TITLE
Switch to LLVM's new pass manager

### ISSUE THREAD

**Issue by kripken** [emscripten-core/emscripten#13427]
LLVM has switched their default to the new pass manager. This caused some regressions for us,

https://github.com/emscripten-core/emscripten/pull/13418#issuecomment-774276340

and a big compile time regression on SIMD tests,

https://github.com/emscripten-core/emscripten/pull/13420

For now, we are staying on the old one,

https://github.com/emscripten-core/emscripten/pull/13426

Hopefully as LLVM stabilizes things over this dev cycle the regressions will go away. If not we may need to live with them.

**Comment by kripken** [comment]
Investigating the new pass manager on the emscripten benchmark suite, we do have a small regression of 20-30 bytes or so pretty much everywhere - likely the dlmalloc issue we saw before - but there are two large changes:

* Havlak is 13% larger (and no change in speed)
* SQLite is 19% smaller and 8% faster

The Havlak changes go away in -O2 or -Os - they are only present in -O3.

Overall, the dlmalloc issue doesn't look like a symptom of a larger problem. Not sure it's worth filing an issue on - we should probably just accept there are some wins and some losses with the new pass manager.

The SIMD compile time slowdown (8 seconds => 7 minutes) is more serious though, and I think we should block switching to the new pass manager on that, I filed https://bugs.llvm.org/show_bug.cgi?id=49102
